pkg/virt-controller/services/template.go

```diff
@@ -1456,6 +1457,7 @@ func (t *templateService) VMIResourcePredicates(vmi *v1.VirtualMachineInstance,
 		vmiCPUArch = t.clusterConfig.GetClusterCPUArch()
 	}
 	memoryOverhead := GetMemoryOverhead(vmi, vmiCPUArch, t.clusterConfig.GetConfig().AdditionalGuestMemoryOverheadRatio)
+	memoryOverhead.Add(netbinding.CalculateMemoryOverhead(vmi.Spec.Domain.Devices.Interfaces, t.clusterConfig.GetNetworkBindings()))
```

Comment:
We need to reconfirm based on the callers of `GetMemoryOverhead` that we do not need to add this overhead at other places.

Reply:
Looks like this needs to be considered also at these callers:
- [Memory hotplug](https://github.com/kubevirt/kubevirt/blob/27976a76f5dfa5ec68f2aba9ea5fab826974805f/pkg/virt-handler/vm.go#L3630).

I do not think (or unsure) it is needed here:
- [Qemu memlock adjustment](https://github.com/kubevirt/kubevirt/blob/bd9ed044896005c3a3ac8b1458d73e4f11809a56/pkg/virt-handler/isolation/detector.go#L161).
  - I would even argue it may not be correct to perform the current calculation like this.
- Similar to the above point, [here](https://github.com/kubevirt/kubevirt/blob/bd9ed044896005c3a3ac8b1458d73e4f11809a56/pkg/virt-handler/isolation/detector.go#L127).

Side note: Why there are two memlock adjustment functions?

Reply:
It looks like qemu memlock adjustment is incorrectly using the guest memory overhead function, because the adjustment should care only about the Guest physical addresses, AKA VM memory only.

Reply:
@orelmisan As @EdDev mentioned, you also need to update the `hotplugMemory` function to take into account the NBP memory overhead

Reply:
@enp0s3 @EdDev Done.